src/utils.ts

```diff
@@ -12,6 +12,18 @@ export const getNextPendingMessage = (batch: PendingMessages): PendingMessage |
         .find((b) => !b.processing)
 }
 
+export const deleteAllPendingMessagesByGroupId = (
```

Comment:
Do we need to preserve the array object? This looks quite weird, all the iterations and splicing :) we need to preserve messages itself but the array structure can be changed right? So how about just returning new array (by filtering)?